File: Bubble Sort/bubble_sort.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

// Bubble Sort 

/* Bubble Sort is the simplest sorting algorithm that works by repeatedly swapping the adjacent elements
if they are in wrong order */

//	Time Complexity of Bubble Sort is O(n^2) 
//	Best Case is O(n) that happens when the array is already sorted

void Bubble_Sort(vector<int>& list)
{
	int len = list.size();

	for (int i = 0; i < len; i++)
	{
		bool swaped = false;
		for (int j = 0; j < len - 1 - i; j++)
		{
			if (list[j] > list[j + 1])
				swap(list[j], list[j + 1]);
				swaped = true;
		}
		if (!swaped) break;
	}
}
int main()
{
	vector<int> list = { 10,5,3,0,12 };
	Bubble_Sort(list);

	for (auto i : list)
		cout << i << " ";
	return 0;
}

File: Bubble Sort/bubble_sort.py
#  Bubble Sort is the simplest sorting algorithm that works by repeatedly swapping the adjacent elements
#  if they are in the wrong order.
#


#  Bubble Sort Function

def bubble_sort(list):
    n = len(list)
    swapped = False

    for i in range(n-1):

        for j in range(n-1-i):

            #   Swap if the element found is greater
            # than the next element
            if list[j] > list[j+1]:
                swapped = True
                (list[j], list[j+1]) = (list[j+1], list[j])

        if not swapped:
            # if we haven't needed to make a single swap, we
            # can just exit the main loop.
            return


if __name__ == "__main__":

    arr = [64, 34, 25, 12, 22, 11, 90]
    bubble_sort(arr)
    print(arr)
